scripts/jenkins/ardana/ansible/list-installed-packages.yml

```diff
@@ -0,0 +1,110 @@
+#
+# (c) Copyright 2019 SUSE LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+---
+
+- import_playbook: _create-vcloud-inventory.yml
+
+- name: Collect list of installed packages in all nodes
+  hosts: "{{ ardana_env }}, cloud_virt_hosts"
+  remote_user: root
+  gather_facts: True
+  vars:
+    task: "deploy"
+    state: "default"
+    ansible_ssh_common_args: >
+      -o ProxyCommand='ssh -o UserKnownHostsFile=/dev/null -o
+      StrictHostKeyChecking=no -W %h:%p -q root@{{ hostvars[ardana_env].ansible_host }}'
+      -o ControlPath='~/.ansible/cp/{{ ardana_env }}-%r@%h:%p'
+
+  pre_tasks:
+    - include_role:
+        name: rocketchat_notify
+      vars:
+        rc_action: "started"
+        rc_state: "Started"
+      when: rc_notify
+
+  tasks:
+    - block:
+        # FIXME this is for testing since we don't have maintupdate jobs
+        - name: FIXME Inject a package
+          zypper:
+            name: "{{ item }}"
+            state: present
+          loop:
+            - "iftop"
+          when: "'after_update' in state"
+
+        - name: Ensure "rpms" artifacts dir exists
+          file:
+            path: "~/rpms"
+            state: "directory"
+
+        - name: Collect list of installed packages in plaintext
+          shell: |
+            rpm -qa | sort > ~/rpms/rpms_{{ inventory_hostname }}_{{ state }}.txt
+          args:
+            warn: False
+            creates: "~/rpms/rpms_{{ inventory_hostname }}_{{ state }}.txt"
+
+        - name: Collect list of installed packages in yaml
+          shell: |
+            rpm -qa --qf "%{NAME}:\n  version: %{VERSION}\n  release: %{RELEASE}\n  disturl: %{DISTURL}\n" \
+            | perl -pe 'chomp;$_.=($.%4)?"\n":"\0"' | sort -z -t' ' -k1 | tr '\0' '\n' \
+            > ~/rpms/rpms_{{ inventory_hostname }}_{{ state }}.yaml
```

Comment:
This is more complicated than it has to be.
You get the same result from
```bash
rpm -qa --qf "%{NAME}:\t  version: %{VERSION}\t  release: %{RELEASE}\t  disturl: %{DISTURL}\n" \
| sort | tr '\t' '\n' \
> ....yaml
```

And you can even drop the tr, if you go for a more JSON-like yaml:
```bash
rpm -qa --qf '"%{NAME}": \{ "version": "%{VERSION}", "release": "%{RELEASE}", "disturl": "%{DISTURL}" \}\n' |
 sort > ....yaml
```

It is still somewhat human-readable:
```yaml
"a2ps": { "version": "4.14", "release": "lp150.1.37", "disturl": "obs://build.opensuse.org/openSUSE:Leap:15.0/standard/9d231eacdd5ca84957a77f837758ec48-a2ps" }
"aaa_base-extras": { "version": "84.87+git20180409.04c9dae", "release": "lp150.8.1", "disturl": "obs://build.opensuse.org/openSUSE:Maintenance:9894/openSUSE_Leap_15.0_Update/43255c1ce72a835dd9154a2a1cec4a85-aaa_base.openSUSE_Leap_15.0_Update" }
```